util/progress/progressui/printer.go

```diff
@@ -47,6 +57,28 @@ func (p *textMux) printVtx(t *trace, dgst digest.Digest) {
 
 	for _, s := range v.statuses {
 		if _, ok := v.statusUpdates[s.ID]; ok {
+			doPrint := true
+
+			if last, ok := p.last[s.ID]; ok {
```

Comment:
` && !s.Completed` ?

Reply:
`s.Completed == nil`, but yes. I was incorrectly thinking that with your previous changes I didn't need to care about completion, but of course I actually do want to print the first (and now only) one unconditionally.